geomstats/discretized_curves_space.py

```diff
@@ -0,0 +1,437 @@
+        srv = srv[index_select, :].reshape(srv_shape)
+
+        return srv
+
+    def square_root_velocity_inverse(self, srv, starting_point):
+        """
+        Retreive a curve from its square root velocity representation
+        and starting point.
+        """
+        if not isinstance(self.embedding_metric, EuclideanMetric):
+            raise AssertionError('The square root velocity inverse is only '
+                                 'implemented for dicretized curves embedded '
+                                 'in a Euclidean space.')
+        if srv.ndim != starting_point.ndim:
+            starting_point = gs.transpose(
+                    np.tile(starting_point, (1, 1, 1)), (1, 0, 2)
+                    )
+        srv_shape = srv.shape
+        srv = gs.to_ndarray(srv, to_ndim=3)
+        n_curves = srv.shape[0]
+        n_sampling_points = srv.shape[1] + 1
+        n_coords = srv.shape[2]
+
+        srv = srv.reshape(n_curves * (n_sampling_points - 1), n_coords)
+        srv_norm = self.embedding_metric.norm(srv)
+        delta_points = 1 / n_sampling_points * srv_norm * srv
+        delta_points = delta_points.reshape(srv_shape)
+        curve = np.concatenate((starting_point, delta_points), -2)
+        curve = np.cumsum(curve, -2)
+
+        return curve
+
+    def exp(self, tangent_vec, base_curve):
+        """
+        Riemannian exponential of a tangent vector wrt to a base curve.
+        """
+        if not isinstance(self.embedding_metric, EuclideanMetric):
+            raise AssertionError('The exponential map is only implemented '
+                                 'for dicretized curves embedded in a '
+                                 'Euclidean space.')
+        base_curve = gs.to_ndarray(base_curve, to_ndim=3)
+        tangent_vec = gs.to_ndarray(tangent_vec, to_ndim=3)
+        n_sampling_points = base_curve.shape[1]
+
+        base_curve_srv = self.square_root_velocity(base_curve)
+
+        tangent_vec_derivative = n_sampling_points * (
+                tangent_vec[:, 1:, :] - tangent_vec[:, :-1, :])
+        base_curve_velocity = n_sampling_points * (
+                base_curve[:, 1:, :] - base_curve[:, :-1, :])
+        base_curve_velocity_norm = self.pointwise_norm(base_curve_velocity,
+                                                       base_curve[:, :-1, :])
+
+        coef_1 = 1 / gs.sqrt(base_curve_velocity_norm)
+        coef_2 = - 1 / (2 * base_curve_velocity_norm ** (5/2)) * \
+            self.pointwise_inner_product(tangent_vec_derivative,
+                                         base_curve_velocity,
+                                         base_curve[:, :-1, :])
+        srv_initial_derivative = gs.transpose(coef_1) * \
+            gs.transpose(tangent_vec_derivative, (2, 1, 0)) + \
+            gs.transpose(coef_2) * gs.transpose(base_curve_velocity, (2, 1, 0))
+        srv_initial_derivative = gs.transpose(srv_initial_derivative,
```

Comment:
There seems to be several transposes: aren't cancelling them out, and could we write it directly without tranposing?